gossip/src/contact_info.rs

```diff
@@ -0,0 +1,293 @@
+#![allow(dead_code)]
+#![allow(unused_variables)]
+#![allow(unused_imports)]
+pub use crate::legacy_contact_info::LegacyContactInfo;
+use {
+    crate::crds_value::MAX_WALLCLOCK,
+    serde::{Deserialize, Deserializer, Serialize, Serializer},
+    solana_sdk::{
+        pubkey::Pubkey,
+        sanitize::{Sanitize, SanitizeError},
+        serde_varint, short_vec,
+    },
+    std::{
+        borrow::Cow,
+        collections::HashMap,
+        net::{IpAddr, SocketAddr},
+        num::TryFromIntError,
+    },
+    thiserror::Error,
+};
+
+const SOCKET_TAG_GOSSIP: u8 = 0;
```

Comment:
Would it be possible to add a gossip quic entry as well? We want to add Quic Gossip support. CC @lijunwangs on the related Quic repair changes for if we need a port allocated for that as well.

Reply:
Instead of adding unused ports now, we can add new ports once the code using those ports is almost ready (could possibly be in the same patch). Even if we add that now it cannot be used yet until `LegacyContactInfo => ContactInfo` migration is done.